draft-packages/table/KaizenDraft/Table/Table.tsx

```diff
@@ -84,36 +85,38 @@ const ratioToPercent = (width?: number) =>
  *        shrink, and basis, due to IE11 compatibility. eg. use "1 1 auto"
  *        instead of just "1".
  */
-type TableHeaderRowCell = React.FunctionComponent<{
-  labelText: string
-  automationId?: string
-  onClick?:
-    | ((e: React.MouseEvent<HTMLButtonElement>) => any)
-    | ((e: React.MouseEvent<HTMLAnchorElement>) => any)
-  href?: string
-  width?: number
-  flex?: string
-  icon?: React.SVGAttributes<SVGSymbolElement>
-  checkable?: boolean
-  checkedStatus?: CheckedStatus
-  onCheck?: (event: React.ChangeEvent<HTMLInputElement>) => any
-  reversed?: boolean
-  /**
-   * This boolean would show a "sort by" icon in the table cell header.
-   * The problem was that the arrow was pointing in the descending direction only.
-   * Please use `sorting` prop instead.
-   * @deprecated
-   */
-  active?: boolean
-  /**
-   * Shows an up or down arrow, to show that the column is sorted.
-   */
-  sorting?: "ascending" | "descending"
-  wrapping?: "nowrap" | "wrap"
-  align?: "start" | "center" | "end"
-  tooltipInfo?: string
-  sortingArrowsOnHover?: "ascending" | "descending" | undefined
-}>
+type TableHeaderRowCell = OverrideClassName<React.FunctionComponent> &
```

Comment:
`OverrideClassName` is a generic which removes `className` from props and adds `classNameOverride` in its place. It is mostly used for when we extend HTML attributes, which don't seem to be the case here.

Options:
- Use `OverrideClassName` on the props within `React.FunctionComponent` (which we should use `React.VFC` instead)
- Just add the prop and not use the generic

I'd probably recommend you pull out the props into their own `interface` too and export it.

Reply:
I've removed the `OverrideClassName` generic - it's just using the prop now

Reply:
@JakePitman really sorry, but I think our changes clashed and I overwrote yours. I think we had the same idea though. I've added a BaseTableProps interface to match the other files.